src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -269,12 +264,63 @@ public void actionPerformed(ActionEvent e) {
             }
         };
         toolbarPanel.setResetSpeedListener(resetSpeedListener);
-
+        final int[] regionWidth = new int[1];
+        final int[] regionHeight = new int[1];
         regionsPanel.addComponentListener(new ComponentAdapter() {
             public void componentResized(ComponentEvent ev) {
+                regionWidth[0] = ev.getComponent().getWidth();
+                regionHeight[0] = ev.getComponent().getHeight();
                 renderRunner.notifyRegionResized(ev.getComponent().getWidth(), ev.getComponent().getHeight());
             }
         });
+        final boolean isReplayFinal = isReplay;
+        regionsPanel.addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                renderRunner.playback.isPaused = true;
+                Snapshot snapshot;
+                if (isReplayFinal) {
+                    snapshot = renderRunner.playback.snapshot;
+                } else {
+                    snapshot = renderRunner.live.snapshot;
+                }
+//                System.out.println(e.getX() + ", " + e.getY());
```

Comment:
I think we should compute the region number _before_ opening a popup. The popup should only open if the region number is valid (in case they haven't clicked on a region).

Reply:
Okay let me see if I can change that.